vms/proposervm/batched_vm.go

```diff
@@ -155,6 +154,30 @@ func (vm *VM) BatchedParseBlock(ctx context.Context, blks [][]byte) ([]snowman.B
 	return blocks, nil
 }
 
+type parseResults struct {
+	block statelessblock.Block
+	err   error
+}
+
+// parseBlocks would parse the given raw blocks into tuples of (Block, error).
+func (vm *VM) parseBlocks(blks [][]byte) []parseResults {
```

Comment:
Would it make sense to move this method off of the `vm` and just pass in the `chainID ids.ID`?

Reply:
sure, done!